### DISCUSSION TITLE
Beautify  feature buggy?

### DISCUSSION THREAD

**Discussion by uri613** [phcode-dev/phoenix#1803] [General]
This code in a PHP file as part of HTML:
`<?= htmlspecialchars($d ?? "", ENT_QUOTES) ?>`

Becomes this when I use beautify feature (breaking the HTML):
```
<?= htmlspecialchars(
            $d ?? "",
            ENT_QUOTES
        ) ?>

```
I've confirmed multiple times and for similar code.
I don't know where else to report this. Until fixed, 'beautify' is a hand-grenade.
Also, think I noticed that beautify changes single quotes to doubles sometimes, just as deadly...

**Comment by abose** [comment]
These changes are syntactically correct and should not break code or html. Are you facing any issues due to this? Tags can be split into multiple lines, beautify splits lines when it sees that the line is too long.

**Reply by uri613** [reply]
Well, seems that for some lines, the problem went away; probably I got confused in the debugging since that was allso a "passing null" issue.
BUT... when beautify breaks up another line for example:
<?= $isFree
            ? "checked"
            : "" ?> />
My editor (Phoenix Code) lists it as a "failed to tokenize" which is at least, annoying, especially when there are a dozen of these things in one of fifty PHP files!
And beautify definitely changes some single quotes to doubles which can be a hazard in middle of some echoed strings..!

**Reply by abose** [reply]
@uri613 Apologies, missed this update. 
Can you give example code that has the issues? without examples, it will be very hard for me to fix the issues.
